### PR TITLE
Go Gin framework Authentication Demo and Fixing error in Guidelines.md file

### PR DESCRIPTION
**Pull request by gagan3012**
#### Checklist

- [X] You have read the [Contribution Guidlines](https://github.com/LoginRadius/developer-authentication-demos/blob/master/guidelines.md) before creating this PR.
- [X] This is not a duplicate entry for any framework or language\

##### Description of change

I have done the following things:

1) Created a Go gin framework demo which uses the go-SDK of login-radius
2) added a Readme file for the demo with all instructions 
3) fixed an error in the guidelines.md file

### CODE REVIEW

File: demos/go/gin/config.env.example [hunk 1]
```diff
@@ -0,0 +1,4 @@
+APIKEY= <..APIKEY..>
+APISECRET= <..APISECRET..>
+ACCESS_TOKEN= <..ACCESS_TOKEN..>
```
Comment (pareek-narendra) [inline demos/go/gin/config.env.example:3]: @gagan3012 
we can not have the token in env file. The value of the token will be available after successful authentication.

File: demos/go/gin/main.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	var errors string
+	respCode := 200
+
+	cfg := lr.Config{
+		ApiKey:    os.Getenv("APIKEY"),
+		ApiSecret: os.Getenv("APISECRET"),
+	}
+	lrclient, err := lr.NewLoginradius(
+		&cfg,
+		map[string]string{"token": os.Getenv("ACCESS_TOKEN")},
```
Comment (pareek-narendra) [inline demos/go/gin/main.go:59]: @gagan3012 
Get the token value from query param i.e `keys := c.Request.URL.Query()`

**Review by pareek-narendra — CHANGES_REQUESTED** [review]

**Comment by pareek-narendra** [comment]
@gagan3012 
You can follow our [guideline ](https://github.com/LoginRadius/developer-authentication-demos/blob/master/guidelines.md) and our [go library](https://www.loginradius.com/docs/developer/sdk-libraries/golang-library/) document for more details.

File: demos/go/gin/config.env.example (hunk 1 above)
Comment (gagan3012) [reply demos/go/gin/config.env.example]: Removed Access_token from env file

File: demos/go/gin/main.go (hunk 2 above)
Comment (gagan3012) [reply demos/go/gin/main.go]: @pareek-narendra fixed both requested changes

**Comment by gagan3012** [comment]
I have made the asked changes please take a look!

**Comment by pareek-narendra** [comment]
@gagan3012 
The demo is not in working condition.
as in you are not passing the status `error` or `success` along with the response/error. The profile will also be in response.data

If you check the existing node js express example 
- In case of `success `
```
output.data = response;
output.message = 'Profile fetched';
output.status = 'success';
```
- In case of `error`
```
output.status = 'error';
output.message = 'an error occoured';
```

**Review by pareek-narendra — CHANGES_REQUESTED** [review]
There was a minor issue in Get and POST Request that I've fixed.
I've also marked `hacktoberfest-accept` label but you can still improve the demo as mentioned below:
- You have used Account API to update the profile but I would recommend You to use authentication API to update the profile using the token as we have in other demos.

**Comment by gagan3012** [comment]
I was looking for the account update function in Authentication api go sdk but it doesn’t exist hence I had to use the account api instead.

**Comment by gagan3012** [comment]
@pareek-narendra 
I have removed the account API please take a look!

File: demos/go/gin/main.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+		respCode = 500
+	} else {
+		var data map[string]interface{}
+		er := json.Unmarshal([]byte(response.Body), &data)
+		if er != nil {
+			//panic(err)
+		}
+		out.Message = "Profile has been updated successfully."
+		out.Status = "success"
+		out.Data = data
```
Comment (pareek-narendra) [inline demos/go/gin/main.go:151]: @gagan3012 Great, now looks good.
Only a minor issue I've found is that profiles are coming in Data Property.
i.e response will be like of this `PutAuthUpdateProfileByToken` function
```
"Body" : {
  "IsPosted":true,
   "Data":{ 
    .....
   }
}
```
So you can check if `IsPosted` property is true and send the profile response in `out.Data`

Get the reference from node js -> express demo also let me know if you need any help in this.

**Comment by gagan3012** [comment]
@pareek-narendra 
I have added the IsPosted check please take a look.

File: demos/go/gin/main.go (hunk 3 above)
Comment (gagan3012) [reply demos/go/gin/main.go]: @pareek-narendra  I have added the check for IsPosted. Thank you!

Comment (pareek-narendra) [reply demos/go/gin/main.go]: @gagan3012 
That is fine and look good but read my last comment properly. I have mentioned one more issue.
`Only a minor issue I've found is that profiles are coming in Data Property.`
Right now you are unmarshalling `response.Body` in line no. 145 but the profile is coming in `response.Body.Data`.
So you need to send the profile in `out.Data`  as the direct profile is handled on the frontend.

Comment (gagan3012) [reply demos/go/gin/main.go]: @pareek-narendra 
Thank you I have fixed it now! Please take a look!

**Review by pareek-narendra — APPROVED** [review]

**Pull request merged**